homeassistant/components/fritz/wrapper.py

```diff
@@ -0,0 +1,97 @@
+"""AVM FRITZ!Box API wrapper."""
+from __future__ import annotations
+
+from functools import partial
+import logging
+from typing import Any
+
+from fritzconnection.core.exceptions import (
+    FritzActionError,
+    FritzActionFailedError,
+    FritzConnectionException,
+    FritzLookUpError,
+    FritzSecurityError,
+    FritzServiceError,
+)
+
+from .common import FritzBoxTools
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AvmWrapper:
+    """Setup AVM wrapper for API calls."""
+
+    def __init__(self, avm_device: FritzBoxTools) -> None:
+        """Init wrapper API class."""
+
+        self._avm_device = avm_device
+
+    def _service_call_action(
+        self,
+        service_name: str,
+        service_suffix: str,
+        action_name: str,
+        **kwargs: Any,
+    ) -> dict | None:
+        """Return service details."""
+
+        if (
+            f"{service_name}{service_suffix}"
+            not in self._avm_device.connection.services
+        ):
+            return None
+
+        try:
+            return self._avm_device.connection.call_action(  # type: ignore[no-any-return]
```

Comment:
Assuming it will always return a `dict[Any, Any]`, you could do something like ...

```suggestion
            result: dict = self._avm_device.connection.call_action(
```

and then return the result

Reply:
Done, was not aware it's the preferred approach.
Thx